ext/standard/image.c

```diff
@@ -1238,38 +1238,38 @@ PHP_FUNCTION(image_type_to_extension)
 
 	switch (image_type) {
 		case IMAGE_FILETYPE_GIF:
-			RETURN_STRING(".gif" + !inc_dot);
+			RETURN_STRING(&".gif"[!inc_dot]);
 		case IMAGE_FILETYPE_JPEG:
-			RETURN_STRING(".jpeg" + !inc_dot);
+			RETURN_STRING(&".jpeg"[!inc_dot]);
 		case IMAGE_FILETYPE_PNG:
-			RETURN_STRING(".png" + !inc_dot);
+			RETURN_STRING(&".png"[inc_dot]);
```

Comment:
This should probably be `RETURN_STRING(&".png"[!inc_dot]);`.

Reply:
true ... typo here.